### PR TITLE
Robustifying default widgets

### PR DESCRIPTION
**Pull request by classner**
Hi qtile team,

this is a small collection of updates to the standard widgets to make them easier to use and more robust (or just work with the current Python version of other libraries). In detail:

**Wallpaper widget**:
* add option to display a fixed string instead of current image name. This can quite clutter the bar.
* remove a bug that required the folder name to end in '/'. Use os.path.join instead.

**khal widget**:
* use `unicode` to make it work with Python2.
* rewrote the event parsing in a more robust way that works with the current version of khal.

**mailfolder widget**:
* did not use user folder expansion for the mail folder.

**Volume widget**:
* expose commands `increase_vol`, `decrease_vol` and `mute` for easy and consistent keybinding in the configs.

### CODE REVIEW

File: libqtile/widget/wallpaper.py [hunk 1]
```diff
@@ -44,7 +48,7 @@ def __init__(self, **config):
         self.set_wallpaper()
 
     def get_path(self, file):
-        return self.directory + file
+        return os.path.join(self.directory, file)
```
Comment (flacjacket) [inline libqtile/widget/wallpaper.py]: You can do `expanduser` here, too, then it is not needed in the defaults

File: libqtile/widget/khal_calendar.py [hunk 2]
```diff
@@ -74,49 +75,50 @@ def poll(self):
...
             except ValueError:
-                date = output[i]
-                caldate = output[i]
-                continue
-            if endtime > now:
-                data = caldate.replace(':', '') + ' ' + output[i]
+                try:
+                    if output[i] == 'Today:':
+                        date = str(now.month) + '/' + str(now.day) + '/' + \
+                                str(now.year)
```
Comment (Azd325) [inline libqtile/widget/khal_calendar.py]: [str.format](https://pyformat.info/) would be nice too

File: libqtile/widget/khal_calendar.py [hunk 3]
```diff
@@ -74,49 +75,50 @@ def poll(self):
...
         # get reminder time in datetime format
         remtime = datetime.timedelta(minutes=self.remindertime)
 
         # parse khal output for the next seven days
         # and get the next event
         args = ['khal', 'agenda', '--days', str(self.lookahead)]
         cal = subprocess.Popen(args, stdout=subprocess.PIPE)
-        output = cal.communicate()[0]
-        output = output.decode()
+        output = unicode(cal.communicate()[0], 'utf-8')
```
Comment (classner) [inline libqtile/widget/khal_calendar.py]: Thanks! Didn't know unicode was not available on Py3... Works!

File: libqtile/widget/khal_calendar.py [hunk 4]
```diff
@@ -74,49 +75,50 @@ def poll(self):
...
                                                   ignoretz=True)
                 endtime = dateutil.parser.parse(date + ' ' + output[i][6:11],
                                                 ignoretz=True)
             except ValueError:
-                date = output[i]
-                caldate = output[i]
-                continue
-            if endtime > now:
-                data = caldate.replace(':', '') + ' ' + output[i]
+                try:
```
Comment (classner) [inline libqtile/widget/khal_calendar.py]: The parsing code I submitted is more robust than the old one and should not have trouble with older khal versions.

File: libqtile/widget/khal_calendar.py (hunk 2 above)
Comment (classner) [inline libqtile/widget/khal_calendar.py]: Fixed. Was on my Laptop w/o linting...

File: libqtile/widget/wallpaper.py (hunk 1 above)
Comment (classner) [inline libqtile/widget/wallpaper.py]: Good suggestion, much cleaner!

**Comment by flacjacket** [comment]
Great thanks for the quick fixes!

**Pull request merged**